templates/sonda.py.j2

```diff
@@ -0,0 +1,206 @@
+  empty_store_responses.labels(node=store_node).inc()
+  consecutive_successful_responses.labels(node=store_node).set(0)
+  return True
+
+
+def send_store_query(rest_address, store_node, encoded_pubsub_topic, encoded_content_topic, timestamp):
+    url = f'{rest_address}/store/v3/messages'
+    params = {
+        'peerAddr': urllib.parse.quote(store_node, safe=''), 
+        'pubsubTopic': encoded_pubsub_topic,
+        'contentTopics': encoded_content_topic, 
+        'includeData': 'true', 
+        'startTime': timestamp
+    }
+    
+    s_time = time.time()
+    
+    try:
+        log_with_utc(f'Sending store request to {store_node}')
+        response = requests.get(url, params=params)
+    except Exception as e:
+      log_with_utc(f'Error sending request: {e}')
+      failed_store_queries.labels(node=store_node, error=str(e)).inc()
+      consecutive_successful_responses.labels(node=store_node).set(0)
+      return False
+
+    elapsed_seconds = time.time() - s_time
+    log_with_utc(f'Response from {rest_address}: status:{response.status_code} [{elapsed_seconds:.4f} s.]')
+
+    if response.status_code != 200:
+        failed_store_queries.labels(node=store_node, error=f'{response.status_code} {response.content}').inc()
+        consecutive_successful_responses.labels(node=store_node).set(0)
+        return False
+
+    # Parse REST response into JSON
+    try:
+        json_response = response.json()
+    except Exception as e:
+        log_with_utc(f'Error parsing response JSON: {e}')
+        failed_store_queries.labels(node=store_node, error="JSON parse error").inc()
+        consecutive_successful_responses.labels(node=store_node).set(0)
+        return False
+
+    # Analyze Store response. Return false if response is incorrect or has an error status
+    if not check_store_response(json_response, store_node, timestamp):
+        return False
+
+    store_query_latency.labels(node=store_node).set(elapsed_seconds)
+    return True
+
+
+def send_store_queries(rest_address, store_nodes, pubsub_topic, content_topic, timestamp):
+    log_with_utc(f'Sending store queries. nodes = {store_nodes} timestamp = {timestamp}')
+    encoded_pubsub_topic = urllib.parse.quote(pubsub_topic, safe='')
+    encoded_content_topic = urllib.parse.quote(content_topic, safe='')
+    
+    for node in store_nodes:
+      send_store_query(rest_address, node, encoded_pubsub_topic, encoded_content_topic, timestamp)
+
+
+def main():
+  log_with_utc(f'Running Sonda with args={args}')
+
+  store_nodes = []
+  if args.store_nodes is not None:
+      store_nodes = [s.strip() for s in args.store_nodes.split(",")]
+  log_with_utc(f'Store nodes to query: {store_nodes}')
+
+  # Start Prometheus HTTP server at port {{ nim_waku_sonda_metrics_port }}
+  start_http_server({{ nim_waku_sonda_metrics_port }})
+
+  node_rest_address = 'http://{{ nim_waku_cont_name }}:{{ nim_waku_rest_port }}'
```

Comment:
Aside from this being not a separate role, why did you turn this file into a template?

I can see this file lives at `apps/sonda/sonda.py` in `nwaku` repo, and that's where it should remain. You've copied it instead of referencing or downloading it. This means every time the script changes you'll have to refactor this template into the new form.

Instead change the script in `nwaku` to accept flags that it needs to accept and simply download it.